### DISCUSSION TITLE
Using isBetween

### DISCUSSION THREAD

**Discussion by rajumb502** [simolus3/drift#1070] [Q&A]
```
  Future<List<Book>> getLimitedBooks(int limit, {int offset = 0})
      =>(select(books)
          ..where((t) => t.srno.isBetweenValues(offset, offset+limit))
              .limit(limit)).get();
```
I was trying to limit the number of rows I fetch from the DB so that I could paginate in the front end. When I try to use the above, I get this error in the IDE: ''error: This expression has a type of 'void' so its value can't be used. (use_of_void_result at ...) for :
```
..where((t) => t.srno.isBetweenValues(offset, offset+limit))
```

The below works without any issues:
```Future<List<Book>> getFavoriteBooks() => (select(books)..where((t) => t.favorite.equals(1))).get();```

Any inputs on what I could be doing wrong? Error comes with both isBetween and isBetweenValues

**Comment by simolus3** [comment]
To fix your example, you would also have to use `..` before `limit`:

```dart
  Future<List<Book>> getLimitedBooks(int limit, {int offset = 0})
      =>(select(books)
          ..where((t) => t.srno.isBetweenValues(offset, offset+limit))
              ..limit(limit)).get();
```

But `limit` has support for offsets builtin, so you don't actually need `isBetweenValues` at all:

```dart
  Future<List<Book>> getLimitedBooks(int limit, {int offset = 0})
      =>(select(books)..limit(limit, offset: offset)).get();
```

I would recommend to always use `orderBy` when using `limit` though. Otherwise the rows returned are not well-defined and might change between sqlite3 versions.

**Reply by rajumb502** [reply]
Thank you @simolus3 . I read somewhere that using a `WHERE` clause causes query to run faster than when using `LIMIT`. Hence the attempt. In my case, `srno` is the PK and is autoincremented, so I assume I don't have to use `orderBy`.
